### PR TITLE
feat: ESLPanel and ESLPanelGroup fallback duration retirement

### PR DESCRIPTION
**Pull request by ala-n**
Closes: #818

### CODE REVIEW

**Review by fshovchko — APPROVED** [review]

File: src/modules/esl-panel-group/core/esl-panel-group.ts [hunk 1]
```diff
@@ -212,14 +208,22 @@ export class ESLPanelGroup extends ESLBaseElement {
...
-      // make sure that browser apply initial height to animate
+      // make sure that browser applies initial height to animate
       afterNextRender(() => {
         this.style.height = `${to}px`;
         this.fallbackAnimate();
       });
     }
   }
 
+  /** Check if transition happens and run afterAnimate step if no transition presented*/
```
Comment (abarmina) [inline src/modules/esl-panel-group/core/esl-panel-group.ts:219]: very minor cosmetic (if you suddenly change something else):
```suggestion
  /** Check if transition happens and run afterAnimate step if no transition presented */
```

**Review by abarmina — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

File: src/modules/esl-panel/core/esl-panel.shape.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export interface ESLPanelTagShape extends ESLToggleableTagShape<ESLPanel> {
   /** Define class(es) to be added during animation after next render ('post-animate' by default) */
   'post-animate-class'?: string;
 
-  /** Define time to clear animation common params (max-height style + classes) ('auto' by default) */
+  /** @deprecated fallback time no longer required */
```
Comment (yadamskaya) [inline src/modules/esl-panel/core/esl-panel.shape.ts:16]:
```suggestion
  /** @deprecated fallback time is no longer required */
```

File: src/modules/esl-panel-group/core/esl-panel-group.ts [hunk 3]
```diff
@@ -212,14 +208,22 @@ export class ESLPanelGroup extends ESLBaseElement {
...
-      // make sure that browser apply initial height to animate
+      // make sure that browser applies initial height to animate
       afterNextRender(() => {
         this.style.height = `${to}px`;
         this.fallbackAnimate();
       });
     }
   }
 
+  /** Check if transition happens and run afterAnimate step if no transition presented */
```
Comment (yadamskaya) [inline src/modules/esl-panel-group/core/esl-panel-group.ts:219]:
```suggestion
  /** Check if transition happens and run afterAnimate step if transition is not presented */
```

File: src/modules/esl-panel/core/esl-panel.ts [hunk 4]
```diff
@@ -102,19 +98,30 @@ export class ESLPanel extends ESLToggleable {
...
+    this.style.setProperty('max-height', `${from}px`);
+    // make sure that browser applies initial height for animation
     afterNextRender(() => {
-      this.style.setProperty('max-height', `${action === 'hide' ? 0 : this._initialHeight}px`);
+      this.style.setProperty('max-height', `${to}px`);
       this.fallbackAnimate();
     });
   }
 
+  /** Check if transition happens and run afterAnimate step if no transition presented*/
```
Comment (yadamskaya) [inline src/modules/esl-panel/core/esl-panel.ts:111]:
```suggestion
  /** Check if transition happens and run afterAnimate step if transition is not presented*/
```

**Review by yadamskaya — APPROVED** [review]

**Review by dshovchko — APPROVED** [review]

File: src/modules/esl-panel-group/core/esl-panel-group.ts [hunk 5]
```diff
@@ -212,14 +208,22 @@ export class ESLPanelGroup extends ESLBaseElement {
...
-      // make sure that browser apply initial height to animate
+      // make sure that browser applies initial height to animate
       afterNextRender(() => {
         this.style.height = `${to}px`;
         this.fallbackAnimate();
       });
     }
   }
 
+  /** Check if transition happens and run afterAnimate step if transition is not presented */
```
Comment (julia-murashko) [inline src/modules/esl-panel-group/core/esl-panel-group.ts:219]: Minor: Cheeks

File: src/modules/esl-panel/core/esl-panel.ts [hunk 6]
```diff
@@ -102,19 +98,30 @@ export class ESLPanel extends ESLToggleable {
...
+    this.style.setProperty('max-height', `${from}px`);
+    // make sure that browser applies initial height for animation
     afterNextRender(() => {
-      this.style.setProperty('max-height', `${action === 'hide' ? 0 : this._initialHeight}px`);
+      this.style.setProperty('max-height', `${to}px`);
       this.fallbackAnimate();
     });
   }
 
+  /** Check if transition happens and run afterAnimate step if transition is not presented*/
```
Comment (julia-murashko) [inline src/modules/esl-panel/core/esl-panel.ts:111]: The same here

**Review by julia-murashko — APPROVED** [review]

**Pull request merged**